File: js/podcast-data.js
// ============================================================================
// ERNEST'S PODCAST - EPISODE DATA
// ============================================================================
// Centralized metadata for all podcast episodes across modules
// All audio files are located in the Podcasts/ directory
// ============================================================================

console.log('🎧 Loading Ernest\'s Podcast episode data...');

export const podcastEpisodes = {
    'emg-introduction': [
        {
            id: 'emg-intro-main',
            title: 'EMG Introduction Fundamentals',
            duration: '101:00',
            audioFile: 'Podcasts/EMG Intro.m4a',
            description: 'Comprehensive introduction to electrodiagnostic medicine fundamentals',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>What is electrodiagnostic medicine?</li>
                    <li>Overview of EMG and NCS techniques</li>
                    <li>Clinical applications and when to order studies</li>
                    <li>What to expect during an EMG/NCS examination</li>
                    <li>Comprehensive fundamentals for beginners</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Electrodiagnostic studies are powerful tools for evaluating nerve and muscle disorders. This comprehensive introduction covers everything you need to get started with EMG and NCS interpretation.</p>
            `
        },
        {
            id: 'emg-terminology',
            title: 'Essential EMG Terminology',
            duration: '28:30',
            audioFile: 'Podcasts/Essential Terminology.m4a',
            description: 'Master the essential vocabulary and definitions used in electrodiagnostic medicine',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Fundamental EMG and NCS terminology</li>
                    <li>Understanding motor unit action potentials (MUAPs)</li>
                    <li>Spontaneous activity patterns and their significance</li>
                    <li>Key concepts: recruitment, conduction velocity, latency</li>
                    <li>Essential definitions for accurate interpretation</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Mastering electrodiagnostic terminology is the foundation for accurate study interpretation. Learn the essential terms and concepts that every EDX physician needs to know.</p>
            `
        }
    ],

    'plexus-anatomy': [
        {
            id: 'plexus-peripheral',
            title: 'Peripheral Nerve Anatomy',
            duration: '75:36',
            audioFile: 'Podcasts/Peripheral Anatomy.m4a',
            description: 'Comprehensive exploration of peripheral nerve anatomy and organization',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Anatomical organization of peripheral nerves</li>
                    <li>Key landmarks and nerve pathways</li>
                    <li>Upper and lower extremity nerve anatomy</li>
                    <li>Clinical correlations with nerve injury patterns</li>
                    <li>Tips for remembering complex anatomy</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Understanding peripheral nerve anatomy is essential for localizing lesions and interpreting electrodiagnostic findings accurately. This comprehensive episode covers the entire peripheral nervous system.</p>
            `
        },
        {
            id: 'plexus-anomalous',
            title: 'Anomalous Anatomy',
            duration: '28:36',
            audioFile: 'Podcasts/Anamolous Anatomy.m4a',
            description: 'Explore anatomical variations and anomalies that can affect EDX interpretation',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Common anatomical variants in peripheral nerves</li>
                    <li>Martin-Gruber anastomosis and its clinical significance</li>
                    <li>Riche-Cannieu anastomosis in the hand</li>
                    <li>How anatomical variations affect NCS interpretation</li>
                    <li>Recognizing anomalies during electrodiagnostic studies</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Anatomical variations are more common than you might think. Learn to recognize anomalous anatomy and avoid misinterpretation of electrodiagnostic studies.</p>
            `
        }
    ],

    'brachial-plexus': [
        {
            id: 'brachial-ep1',
            title: 'Plexopathies',
            duration: '25:00',
            audioFile: 'Podcasts/Plexopathies.m4a',
            description: 'Ernest explores brachial and lumbosacral plexus pathology',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Brachial and lumbosacral plexus anatomy review</li>
                    <li>Clinical syndromes (Erb's palsy, Klumpke's palsy, traumatic injuries)</li>
                    <li>Electrodiagnostic evaluation of plexopathies</li>
                    <li>Differential diagnosis: plexus vs. root vs. nerve</li>
                    <li>Common causes: trauma, radiation, idiopathic inflammation</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Plexopathies can be challenging to diagnose. Learn the clinical and electrodiagnostic features that distinguish plexus lesions from other localizations.</p>
            `
        }
    ],

    'radiculopathy': [
        {
            id: 'radiculopathy-ep1',
            title: 'Radiculopathy Pathophysiology',
            duration: '19:00',
            audioFile: 'Podcasts/Radiculopathy Pathophysiology.m4a',
            description: 'Ernest explains the pathophysiology of nerve root compression',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Mechanisms of nerve root compression and injury</li>
                    <li>Classic radiculopathy patterns (C5-T1, L4-S1)</li>
                    <li>EMG findings: denervation and reinnervation patterns</li>
                    <li>Differentiating radiculopathy from peripheral neuropathy and plexopathy</li>
                    <li>Clinical presentation and physical exam findings</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Radiculopathy is one of the most common reasons for EMG referral. Learn to recognize the patterns and correlate with imaging findings for accurate diagnosis.</p>
            `
        }
    ],

    'neuropathy-pathophysiology': [
        {
            id: 'neuropathy-poly',
            title: 'Polyneuropathies',
            duration: '32:12',
            audioFile: 'Podcasts/Polyneuropathies.m4a',
            description: 'Comprehensive exploration of polyneuropathy patterns and mechanisms',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Axonal vs. demyelinating polyneuropathies</li>
                    <li>Length-dependent vs. non-length-dependent patterns</li>
                    <li>Common causes: diabetes, toxins, hereditary conditions</li>
                    <li>NCS and EMG patterns in polyneuropathies</li>
                    <li>Diagnostic approach and differential diagnosis</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Polyneuropathies are among the most common peripheral nerve disorders. Master the pattern recognition and diagnostic approach for accurate classification and treatment.</p>
            `
        },
        {
            id: 'neuropathy-mono',
            title: 'Mononeuropathy Pathophysiology',
            duration: '64:24',
            audioFile: 'Podcasts/Mononeuropathy pathophysiology.m4a',
            description: 'Detailed examination of focal neuropathies and mononeuropathy multiplex',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Focal entrapment neuropathies: carpal tunnel, cubital tunnel, and more</li>
                    <li>Mononeuropathy multiplex patterns and causes</li>
                    <li>Compression, ischemia, and inflammation mechanisms</li>
                    <li>Electrodiagnostic localization techniques</li>
                    <li>Clinical localization and differential diagnosis strategies</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Understanding mononeuropathy pathophysiology is essential for accurate localization. Learn to distinguish focal compression from systemic causes and recognize mononeuropathy multiplex patterns.</p>
            `
        }
    ],

    'basic-patterns': [
        {
            id: 'patterns-ep1',
            title: 'Basic Pattern Recognition',
            duration: '29:00',
            audioFile: 'Podcasts/Basic Patter Recognition.m4a',
            description: 'Ernest teaches systematic EMG interpretation and pattern recognition',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Systematic approach to EMG/NCS interpretation</li>
                    <li>Classic patterns: CTS, cubital tunnel, fibular neuropathy</li>
                    <li>Radiculopathy patterns and myotomal distributions</li>
                    <li>Plexopathy vs. multiple mononeuropathies</li>
                    <li>Putting it all together: clinical correlation and diagnosis</li>
                    <li>Common pitfalls and how to avoid misinterpretation</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Pattern recognition is a skill that develops with practice. Learn a systematic approach to interpreting EMG/NCS studies and recognizing common patterns to build your diagnostic confidence.</p>
            `
        }
    ],

    'neuropathy-myopathy': [
        {
            id: 'neuro-myo-ep1',
            title: 'Neuropathy vs. Myopathy',
            duration: '32:00',
            audioFile: 'Podcasts/Neuropathy vs. Myopathy.m4a',
            description: 'Ernest explains how to differentiate nerve and muscle disorders using clinical and EDX features',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Clinical clues: distribution of weakness (proximal vs. distal)</li>
                    <li>NCS findings: normal in myopathy, abnormal in neuropathy</li>
                    <li>Needle EMG: myopathic vs. neuropathic changes in detail</li>
                    <li>Motor neuron disease: the great imitator with mixed features</li>
                    <li>Inflammatory myopathies and their EDX signatures</li>
                    <li>Practical diagnostic algorithms and clinical correlation</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Distinguishing neuropathy from myopathy is essential for proper patient management. Use clinical history, NCS, and needle EMG together for accurate diagnosis and appropriate treatment planning.</p>
            `
        }
    ],

    'simple-reports': [
        {
            id: 'reports-ep1',
            title: 'Basic Report Writing',
            duration: '23:06',
            audioFile: 'Podcasts/Basic Report Writing.m4a',
            description: 'Ernest teaches the art of clear, effective electrodiagnostic report writing',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>AANEM reporting guidelines overview</li>
                    <li>Essential elements: patient data, NCS, EMG, summary</li>
                    <li>Writing clear interpretations and recommendations</li>
                    <li>Common mistakes to avoid</li>
                    <li>Tips for efficient and professional report generation</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>A well-written report communicates your findings clearly to referring physicians and becomes part of the permanent medical record. Take time to craft quality reports that accurately convey your electrodiagnostic findings.</p>
            `
        }
    ],

    'extra-topics': [
        {
            id: 'extra-als',
            title: 'ALS and Mimics',
            duration: '26:00',
            audioFile: 'Podcasts/ALS and Mimics.m4a',
            description: 'Ernest explores motor neuron disease and conditions that can mimic ALS',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Pathophysiology of amyotrophic lateral sclerosis (ALS)</li>
                    <li>Clinical presentation: upper and lower motor neuron signs</li>
                    <li>Electrodiagnostic features: widespread denervation and fasciculations</li>
                    <li>ALS mimics: multifocal motor neuropathy, Kennedy's disease, cervical myelopathy</li>
                    <li>Diagnostic criteria and role of EDX in confirming diagnosis</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>ALS is a devastating diagnosis that must be made carefully. Learn to recognize the classic features and rule out treatable mimics through thorough clinical and electrodiagnostic evaluation.</p>
            `
        },
        {
            id: 'extra-blink',
            title: 'The Blink Reflex Explained',
            duration: '26:00',
            audioFile: 'Podcasts/The Blink Reflex explained.m4a',
            description: 'Ernest explains the blink reflex study and its clinical applications',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Anatomy of the blink reflex pathway (trigeminal and facial nerves)</li>
                    <li>R1 and R2 components: what they mean and how to measure them</li>
                    <li>Technical aspects: stimulation, recording, and interpretation</li>
                    <li>Clinical applications: facial nerve disorders, brainstem lesions</li>
                    <li>Troubleshooting and common artifacts</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>The blink reflex is a specialized study for evaluating the trigeminal and facial nerve pathways. Learn when to use it and how to interpret the results for optimal diagnostic yield.</p>
            `
        },
        {
            id: 'extra-nmj',
            title: 'Neuromuscular Junction Disorders',
            duration: '29:00',
            audioFile: 'Podcasts/Neuromuscular junction disorders.m4a',
            description: 'Ernest explores disorders affecting the neuromuscular junction',
            transcript: 'Episode transcript will appear here once available...',
            showNotes: `
                <h4>In This Episode:</h4>
                <ul>
                    <li>Pathophysiology: presynaptic vs. postsynaptic NMJ disorders</li>
                    <li>Myasthenia gravis and Lambert-Eaton syndrome comparison</li>
                    <li>Clinical presentation: weakness patterns and associated symptoms</li>
                    <li>Repetitive nerve stimulation: decremental vs. incremental responses</li>
                    <li>Single fiber EMG findings in NMJ disorders</li>
                    <li>Antibody testing and clinical correlation</li>
                    <li>Treatment implications and prognosis</li>
                </ul>
                <h4>Key Takeaways:</h4>
                <p>Neuromuscular junction disorders require specialized electrodiagnostic techniques for diagnosis. Master the clinical and EDX features that distinguish these treatable conditions from other causes of weakness.</p>
            `
        }
    ]
};

// Helper function to get episodes for a specific module
export function getModuleEpisodes(moduleId) {
    return podcastEpisodes[moduleId] || [];
}

// Get all available module IDs with podcasts
export function getPodcastModules() {
    return Object.keys(podcastEpisodes);
}

console.log('✅ Ernest\'s Podcast episode data loaded -', getPodcastModules().length, 'modules configured');

File: EMG_Mastery_Flutter/lib/data/podcast_data.dart
import 'models/podcast_model.dart';

class PodcastData {
  static final Map<String, List<PodcastEpisode>> episodes = {
    'emg-introduction': [
      PodcastEpisode(
        id: 'emg-intro-main',
        title: 'EMG Introduction Fundamentals',
        duration: '52:27',
        audioFile: 'Podcasts/EMG Intro.m4a',
        description:
            'Comprehensive introduction to electrodiagnostic medicine fundamentals',
      ),
      PodcastEpisode(
        id: 'emg-terminology',
        title: 'Essential EMG Terminology',
        duration: '64:49',
        audioFile: 'Podcasts/Essential Terminology.m4a',
        description: 'Master the essential vocabulary and definitions',
      ),
    ],
    'plexus-anatomy': [
      PodcastEpisode(
        id: 'plexus-peripheral',
        title: 'Peripheral Nerve Anatomy',
        duration: '39:07',
        audioFile: 'Podcasts/Peripheral Anatomy.m4a',
        description: 'Comprehensive exploration of peripheral nerve anatomy',
      ),
      PodcastEpisode(
        id: 'plexus-anomalous',
        title: 'Anomalous Anatomy',
        duration: '14:47',
        audioFile: 'Podcasts/Anamolous Anatomy.m4a',
        description: 'Explore anatomical variations and anomalies',
      ),
      PodcastEpisode(
        id: 'plexus-plexopathies',
        title: 'Plexopathies Deep Dive',
        duration: '13:28',
        audioFile: 'Podcasts/Plexopathies.m4a',
        description:
            'Detailed exploration of brachial and lumbosacral plexus disorders',
      ),
    ],
    'radiculopathy': [
      PodcastEpisode(
        id: 'radiculopathy-ep1',
        title: 'Radiculopathy Pathophysiology',
        duration: '10:17',
        audioFile: 'Podcasts/Radiculopathy.m4a',
        description:
            'Ernest explains the pathophysiology of nerve root compression',
      ),
    ],
    'neuropathy-pathophysiology': [
      PodcastEpisode(
        id: 'neuropathy-poly',
        title: 'Polyneuropathies',
        duration: '16:02',
        audioFile: 'Podcasts/Polyneuropathies.m4a',
        description: 'Comprehensive exploration of polyneuropathy patterns',
      ),
      PodcastEpisode(
        id: 'neuropathy-mono',
        title: 'Mononeuropathy Pathophysiology',
        duration: '33:22',
        audioFile: 'Podcasts/Mononeuropathy pathophysiology.m4a',
        description: 'Detailed examination of focal neuropathies',
      ),
      PodcastEpisode(
        id: 'neuropathy-main',
        title: 'Neuropathy Pathophysiology',
        duration: '33:22',
        audioFile: 'Podcasts/Neuropathy Pathophysiology.m4a',
        description:
            'Foundational pathophysiology of peripheral nerve disorders',
      ),
      PodcastEpisode(
        id: 'neuropathy-myo-vs-neuro',
        title: 'Myopathy vs. Neuropathy',
        duration: '17:09',
        audioFile: 'Podcasts/Myopathy vs. Neuropathy.m4a',
        description: 'Key clinical and electrodiagnostic differentiators',
      ),
    ],
    'basic-patterns': [
      PodcastEpisode(
        id: 'patterns-ep1',
        title: 'Basic Pattern Recognition',
        duration: '14:59',
        audioFile: 'Podcasts/Basic Patter Recognition.m4a',
        description: 'Ernest teaches systematic EMG interpretation',
      ),
      PodcastEpisode(
        id: 'patterns-report-writing',
        title: 'Basic Report Writing',
        duration: '11:56',
        audioFile: 'Podcasts/Basic Report Writing.m4a',
        description: 'Master the art of clear and concise EDX reporting',
      ),
      PodcastEpisode(
        id: 'patterns-interpretation',
        title: 'EMG Interpretation',
        duration: '14:59',
        audioFile: 'Podcasts/EMG interpretation.m4a',
        description: 'Systematic approach to interpreting complex EMG signals',
      ),
    ],
    'extra-topics': [
      PodcastEpisode(
        id: 'extra-als',
        title: 'ALS and Mimics',
        duration: '13:54',
        audioFile: 'Podcasts/ALS and mimics.m4a',
        description: 'Ernest explores motor neuron disease and mimics',
      ),
      PodcastEpisode(
        id: 'extra-blink',
        title: 'The Blink Reflex Explained',
        duration: '14:14',
        audioFile: 'Podcasts/The Blink Reflex explained.m4a',
        description: 'Ernest explains the blink reflex study',
      ),
      PodcastEpisode(
        id: 'extra-blink-alt',
        title: 'Blink Reflex Essentials',
        duration: '14:14',
        audioFile: 'Podcasts/Blink Reflex.m4a',
        description: 'Core principles of the electrodiagnostic blink reflex',
      ),
      PodcastEpisode(
        id: 'extra-nmj',
        title: 'Neuromuscular Junction Disorders',
        duration: '16:00',
        audioFile: 'Podcasts/Neuromuscular junction disorders.m4a',
        description: 'Ernest explores disorders affecting the NMJ',
      ),
    ],
    'edx-series': [
      PodcastEpisode(
        id: 'edx-1',
        title: 'EDX Series: Episode 1',
        duration: '31:44',
        audioFile: 'Podcasts/EDX_1.m4a',
        description: 'A fundamental primer on electrodiagnostic medicine.',
      ),
      PodcastEpisode(
        id: 'edx-2',
        title: 'EDX Series: Episode 2',
        duration: '28:29',
        audioFile: 'Podcasts/EDX_2.m4a',
        description:
            'Technical overview of the neuromuscular system and nerve injuries.',
      ),
      PodcastEpisode(
        id: 'edx-3',
        title: 'EDX Series: Episode 3',
        duration: '36:29',
        audioFile: 'Podcasts/EDX_3.m4a',
        description: 'Overview of EDX instrumentation and NCS principles.',
      ),
      PodcastEpisode(
        id: 'edx-4',
        title: 'EDX Series: Episode 4',
        duration: '43:10',
        audioFile: 'Podcasts/EDX_4.m4a',
        description: 'Details various nerve conduction studies (NCS).',
      ),
      PodcastEpisode(
        id: 'edx-5',
        title: 'EDX Series: Episode 5',
        duration: '35:24',
        audioFile: 'Podcasts/EDX_5.m4a',
        description: 'Diagnostic utility of EDX studies and needle EMG.',
      ),
      PodcastEpisode(
        id: 'edx-6',
        title: 'EDX Series: Episode 6',
        duration: '34:56',
        audioFile: 'Podcasts/EDX_6.m4a',
        description: 'Clinical guide for interpreting EMG results.',
      ),
      PodcastEpisode(
        id: 'edx-7',
        title: 'EDX Series: Episode 7',
        duration: '29:57',
        audioFile: 'Podcasts/EDX_7.m4a',
        description: 'Diagnostic framework for identifying nerve injuries.',
      ),
      PodcastEpisode(
        id: 'edx-8',
        title: 'EDX Series: Episode 8',
        duration: '28:28',
        audioFile: 'Podcasts/EDX_8.m4a',
        description: 'Clinical guide to plexopathies and nerve entrapments.',
      ),
      PodcastEpisode(
        id: 'edx-9',
        title: 'EDX Series: Episode 9',
        duration: '37:25',
        audioFile: 'Podcasts/EDX_9.m4a',
        description:
            'Comprehensive overview of upper extremity nerve pathologies.',
      ),
      PodcastEpisode(
        id: 'edx-10',
        title: 'EDX Series: Episode 10',
        duration: '32:36',
        audioFile: 'Podcasts/EDX_10.m4a',
        description: 'Clinical guide to mononeuropathies of the limbs.',
      ),
      PodcastEpisode(
        id: 'edx-11',
        title: 'EDX Series: Episode 11',
        duration: '37:13',
        audioFile: 'Podcasts/EDX_11.m4a',
        description: 'Anatomy and pathology of lower extremity nerves.',
      ),
      PodcastEpisode(
        id: 'edx-12',
        title: 'EDX Series: Episode 12',
        duration: '28:43',
        audioFile: 'Podcasts/EDX_12.m4a',
        description: 'Clinical overview of peripheral polyneuropathy.',
      ),
      PodcastEpisode(
        id: 'edx-13',
        title: 'EDX Series: Episode 13',
        duration: '40:27',
        audioFile: 'Podcasts/EDX_13.m4a',
        description:
            'Clinical guide for diagnosing neuromuscular junction (NMJ) disorders.',
      ),
      PodcastEpisode(
        id: 'edx-14',
        title: 'EDX Series: Episode 14',
        duration: '34:38',
        audioFile: 'Podcasts/EDX_14.m4a',
        description:
            'Comprehensive clinical guide for diagnosing and classifying myopathies.',
      ),
      PodcastEpisode(
        id: 'edx-15',
        title: 'EDX Series: Episode 15',
        duration: '29:43',
        audioFile: 'Podcasts/EDX_15.m4a',
        description: 'Clinical guide to motor neuron diseases (MND).',
      ),
    ],
  };

  static List<PodcastEpisode> get allEpisodes {
    return episodes.values.expand((list) => list).toList();
  }

  static List<PodcastEpisode> getEpisodesByModule(String moduleId) {
    return episodes[moduleId] ?? [];
  }
}
